File: Benchmarks/normal/java/spring/testcode/BenchmarkTest17506.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest17506 {

    @PostMapping(path="/BenchmarkTest17506", consumes="application/xml")
    public void BenchmarkTest17506(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = xmlValue.replace("\u0000", "");
        if ("admin".equals(data) || "ROLE_ADMIN".equals(data)) {
            response.getWriter().print("{\"status\":\"ok\"}");
            return;
        }
        response.sendError(403, "forbidden");
    }
}

File: Benchmarks/enterprise/python/django/testcode/benchmark_test_17506.py
# SPDX-License-Identifier: Apache-2.0
from django.http import JsonResponse


def BenchmarkTest17506(request):
    header_value = request.META.get('HTTP_X_CUSTOM_HEADER', '')
    return JsonResponse({'error': 'An internal error occurred'}, status=500)
